app/controllers/assignments_controller.rb

```diff
@@ -74,7 +90,7 @@ def create
       @assignment.available = @assignment.due_date
     end
 
-    if set_lateness_config and @assignment.save and set_grader
+    if @assignment.save
```

Comment:
These did two things:
 - Interpreting the form data, which Rails should handle.
 - Validation, which should be handled in the model.